src/Hazelcast.Net/Clustering/ClusterState.cs

```diff
@@ -51,6 +51,10 @@ public ClusterState(HazelcastOptions options, string clusterName, string clientN
 
             _stateChangeQueue = new StateChangeQueue(loggerFactory);
 
+            // FIXME failover should be created outside of the state class
```

Comment:
Yeap, `AddressProvide` is reason why `Failover` here.

Reply:
yea... that FIXME should be a TODO really - anyways after we're done with compact I'll cleanup the FIXMEs. let's leave that one here for now.